RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
@Bzngr when a class has a property getter marked with a `VB_UserMemId = -4` attribute, it means the class is a collection that can be iterated with a `For Each` loop. So we pick up that attribute already, but the point is that a `Let` /value assignment to an instance of such a class can't work:

    Dim foo As Collection
    Set foo = New Collection
    [Let] foo[.Item] = 42 'illegal

So a `Collection` class has a *default property*, but it cannot be assigned a value, because it's parameterized.

In contrast, a `Range` has a *default property*, which can very well be legally assigned with a value:

    Dim foo As Range
    Set foo = ActiveSheet.Range("A1")
    [Let] foo[.Value] = 42 ' totally legal

The inspection is currently accounting for types with a default property, but not for types with an *enumerator* attribute (that's the `UserMemId = -4` attribute); hence it correctly ignores a value assignment to a `Range`, but also incorrectly ignores a value assignment to, say, a `Collection`.

Our `ClassModuleDeclaration` class therefore needs to expose a property that tells us whether the class has that `UserMemId = -4` attribute on one of its members; this will make the bugs in both inspections much easier to fix.